src/main/java/pl/patrycja/ox/ui/UIFactory.java

```diff
@@ -1,13 +1,32 @@
 package pl.patrycja.ox.ui;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * A class UIFactory return a new object UI.
  *
  * @author Patrycja Hyjek
  */
 public class UIFactory {
 
-    public static UI set(boolean demo) {
-        return demo ? new FileUI() : new ConsoleUI();
+    private Map<Integer, UI> kindsOfUi;
+
+    public UIFactory() {
+        this.kindsOfUi = setUi();
+    }
+
+    public UI getUiFor(int inputArraySize) {
+        UI ui = kindsOfUi.get(inputArraySize);
+        return ui;
+    }
+
+    private Map<Integer, UI> setUi() {
```

Comment:
👍  good that this is not a switch this time but do you need 3 times `new ConsoleUI()`

Reply:
Probably I do not need it. I will think about it through.